syntax = "proto3";
package services;

// python -m grpc_tools.protoc -I. --python_out=../services --pyi_out=../services --grpc_python_out=../services analyzer.proto

//import "google/protobuf/timestamp.proto";

service DomainAnalysisService {
  rpc GetFullScoring(Domain) returns (FullDomainScoring);
}

message Domain {
  string name = 1;
}

message FullDomainScoring {
  float finalScore = 1;
  float semanticScore = 2;
  float resourceScore = 3;
  float dgaScore = 4;

  ScoreFlag tag = 5;
}

enum ScoreFlag {
  DOMAIN_SCORE_BENIGN = 0;
  DOMAIN_SCORE_DUBIOUS = 1;
  DOMAIN_SCORE_SUSPICIOUS = 2;
  DOMAIN_SCORE_MALICIOUS = 3;
}